### PR TITLE
[ENH] LSI Color Words by Weight Sign

### PR DESCRIPTION
**Pull request by nikicc**
##### Issue

Words in LSI can have either positive or negative weight and one could not distinguish between them.

##### Description of changes
Topic modelling widget colors LSI words with green (positive weight) and red (negative weight). Also, the top three words are removed from the feature names, since they can correspond to negative weights and are hence misleading.

##### Includes
- [X] Code changes
- [ ] Tests
- [x] Documentation

### CODE REVIEW

**Comment by nikicc** [comment]
Tests will fail until [NLTK resolves its issues](https://stackoverflow.com/questions/45318066/getting-405-while-trying-to-download-nltk-dta). I suggest that tests are run locally before merging this to assure all is OK.

File: orangecontrib/text/widgets/owtopicmodeling.py [hunk 1]
```diff
@@ -229,13 +231,26 @@ def send_topic_by_id(self, topic_id=None):
...
+        self.weights = weights
+        self.color_by_wights_sign = color_by_wights_sign
 
         self.setText(0, '{:d}'.format(topic_id + 1))
-        self.setText(1, ', '.join(words))
+        self.setText(1, ', '.join(self._color(word, weight)
+                                  for word, weight in zip(words, weights)))
+
+    def _color(self, word, weight):
+        if self.color_by_wights_sign:
```
Comment (ajdapretnar) [inline orangecontrib/text/widgets/owtopicmodeling.py]: color_by_w**e**ights ?

**Comment by ajdapretnar** [comment]
@nikicc Could you please rebase?

The widget looks nice. I like to chosen color hues, they work well. Could we make this coloring available also in the report?

I will prepare documentation for this PR and add a commit.

File: orangecontrib/text/widgets/owtopicmodeling.py (hunk 1 above)
Comment (nikicc) [inline orangecontrib/text/widgets/owtopicmodeling.py]: Tnx for noticing. It should be fixed now.

**Comment by nikicc** [comment]
@ajdapretnar I added colors to report and also included a small fix for `All Topics` output.

**Comment by ajdapretnar** [comment]
Thanks! I am just about finished with the documentation, I'll make a PR asap.

**Comment by ajdapretnar** [comment]
@nikicc You can check the documentation if you got a spare minute. Otherwise, I'm ready to merge this.

File: doc/widgets/topicmodelling.rst [hunk 2]
```diff
@@ -53,21 +57,28 @@ Description
...
+
+We then select the first topic and display the most frequent words in the topic in :doc:`Word Cloud <wordcloud>`. We also connected **Preprocess Text** to **Word Cloud** in order to be able to output selected documents. Now we can select a specific word in the word cloud, say *little*. It will be colored red and also highlighted in the word list on the left.
+
+Now we can observe all the documents containing the word *little* in :doc:`Corpus Viewer <corpusviewer>`.
+
+In the second example we will observe topic distribution across words and documents. Connect **Topic Modelling** to **Heat Map**. Ensure the link is set to *All Topics** - *Data*. **Topic Modelling** will output a matrix of topic weights by words from text (more precisely, tokens).
 
-Now we can observe all the documents containing the word *polly* in :doc:`Corpus Viewer <corpusviewer>`.
+We can observe the output in a **Data Table**. Tokens are in rows and retrieved topics in colums. Values represent how much a word in represented in a topic.
```
Comment (nikicc) [inline doc/widgets/topicmodelling.rst]: Values represent how much a word **is** represented in a topic.

File: doc/widgets/topicmodelling.rst [hunk 3]
```diff
@@ -53,21 +57,28 @@ Description
...
+LSI provides both positive and negative weights per topic. A positive weight means the word is highly representative of a topic, while a negative weight means the word is highly unrepresentative of a topic (the less it occurs in a text, the more likely the topic). Positive words are colored green and negative words are colored red.
+
+We then select the first topic and display the most frequent words in the topic in :doc:`Word Cloud <wordcloud>`. We also connected **Preprocess Text** to **Word Cloud** in order to be able to output selected documents. Now we can select a specific word in the word cloud, say *little*. It will be colored red and also highlighted in the word list on the left.
+
+Now we can observe all the documents containing the word *little* in :doc:`Corpus Viewer <corpusviewer>`.
+
+In the second example we will observe topic distribution across words and documents. Connect **Topic Modelling** to **Heat Map**. Ensure the link is set to *All Topics** - *Data*. **Topic Modelling** will output a matrix of topic weights by words from text (more precisely, tokens).
```
Comment (nikicc) [inline doc/widgets/topicmodelling.rst]: I would rephrase `observe topic distribution across words and documents`. Perhaps `In the second example, we will look at the correlation between words (documents) and topics.`?

Comment (nikicc) [inline doc/widgets/topicmodelling.rst]: Also, there's one too many star in `*All Topics**`.

**Comment by ajdapretnar** [comment]
I addressed the comments, should be ok now.

**Comment by nikicc** [comment]
Then, as I'm concerned, you can merge 😄
